### PR TITLE
[API-19485] - Node 18 upgrade

### PR DESCRIPTION
**Pull request by gileswells**
### Description

This takes the previous PR merge for Node 18 and applies it to master. The previous branch was setup to give a static target vs a constantly changing one if we'd merged directly to master.

This includes:
Node upgrade to 18.14.2
NPM upgrade to 9.5.0
React upgrade to 17.0.2
Cypress upgrade to 12.8.1
and many other package upgrades

### Process

- [ ] Tests added or updated for change
- [ ] Docs updated
  - [ ] If you have made structural changes to the site, the [Developer Portal Content Types](https://community.max.gov/x/qUwOg) docs have been updated
  - [ ] If you have made/confirmed any decisions about user interactions and accessibility, the [Accessible Component Design](https://community.max.gov/x/xS8mgg) docs have been updated
- [ ] Accessibility concerns have been considered and addressed
  
  - [ ] If you've made any UI changes, you've run an axe check using the axe browser extension that reported no new issues

### CODE REVIEW

File: .github/workflows/npm-checks.yml [hunk 1]
```diff
@@ -63,20 +63,20 @@ jobs:
...
+          node-version: 18
+      # - id: cache_node_modules
+      #   name: Cache node modules
+      #   uses: actions/cache@v2
+      #   env:
+      #     cache-name: developer-portal-node-modules
+      #     node-version: ${{ needs.install.outputs.node_version }}
+      #   with:
+      #     path: node_modules
+      #     key: ${{ runner.os }}-${{ env.node-version }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
```
Comment (imcgonigle) [inline .github/workflows/npm-checks.yml:75]: Was this comment left here intentionally?

File: .github/workflows/npm-checks.yml [hunk 2]
```diff
@@ -63,20 +63,20 @@ jobs:
...
 
       - name: Install dependencies
-        if: steps.cache_node_modules.outputs.cache-hit != 'true'
-        run: npm ci
+        # if: steps.cache_node_modules.outputs.cache-hit != 'true'
```
Comment (imcgonigle) [inline .github/workflows/npm-checks.yml:78]: Was this comment left here intentionally?

File: .github/workflows/npm-checks.yml [hunk 3]
```diff
@@ -103,20 +103,20 @@ jobs:
...
+          node-version: 18
+      # - id: cache_node_modules
+      #   name: Cache node modules
+      #   uses: actions/cache@v2
+      #   env:
+      #     cache-name: developer-portal-node-modules
+      #     node-version: ${{ needs.install.outputs.node_version }}
+      #   with:
+      #     path: node_modules
+      #     key: ${{ runner.os }}-${{ env.node-version }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
```
Comment (imcgonigle) [inline .github/workflows/npm-checks.yml:115]: Was this comment left here intentionally?

File: .github/workflows/npm-checks.yml [hunk 4]
```diff
@@ -103,20 +103,20 @@ jobs:
...
 
       - name: Install dependencies
-        if: steps.cache_node_modules.outputs.cache-hit != 'true'
-        run: npm ci
+        # if: steps.cache_node_modules.outputs.cache-hit != 'true'
```
Comment (imcgonigle) [inline .github/workflows/npm-checks.yml:118]: Was this comment left here intentionally?

File: .github/workflows/pr-builds.yml [hunk 5]
```diff
@@ -43,19 +43,19 @@ jobs:
...
-          key: ${{ runner.os }}-${{ env.node-version }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+      # - id: cache_node_modules
+      #   name: Cache node modules
+      #   uses: actions/cache@v2
+      #   env:
+      #     cache-name: developer-portal-node-modules
+      #     node-version: ${{ steps.node_version.outputs.node_version }}
+      #   with:
+      #     path: node_modules
+      #     key: ${{ runner.os }}-${{ env.node-version }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
```
Comment (imcgonigle) [inline .github/workflows/pr-builds.yml:54]: Was this comment left here intentionally?

File: .github/workflows/pr-builds.yml [hunk 6]
```diff
@@ -69,20 +69,20 @@ jobs:
...
+          node-version: 18
+      # - id: cache_node_modules
+      #   name: Cache node modules
+      #   uses: actions/cache@v2
+      #   env:
+      #     cache-name: developer-portal-node-modules
+      #     node-version: ${{ needs.install.outputs.node_version }}
+      #   with:
+      #     path: node_modules
+      #     key: ${{ runner.os }}-${{ env.node-version }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
```
Comment (imcgonigle) [inline .github/workflows/pr-builds.yml:81]: Was this comment left here intentionally?

File: src/containers/documentation/swaggerPlugins/InfoContainer.tsx [hunk 7]
```diff
@@ -23,19 +23,23 @@ const InfoContainer: React.FunctionComponent<InfoContainerProps> = (
...
-    url.pathname = metadata?.path ?? url.pathname;
-    urlOutput = url.href;
+    try {
+      const url = new URL(specUrl);
+      const metadata = versionMetadata?.find(obj => obj.sf_path === url.pathname);
+      url.pathname = metadata?.path ?? url.pathname;
+      urlOutput = url.href;
+    } catch (e: unknown) {
+      // This shouldn't be needed but GitHub Actions errors for some reason on this part.
+    }
```
Comment (imcgonigle) [inline src/containers/documentation/swaggerPlugins/InfoContainer.tsx:42]: I feel like we should be doing something here so the app doesn't just swallow the error. Maybe even a console.log so we have something to lead us back here if there are errors in the future

**Review by imcgonigle — DISMISSED** [review]

**Comment by gileswells** [comment]
@imcgonigle So the commented cache lines were from an issue that was happening where the change from node 12 > 18 didn't properly bust the cache when we had some places used `--legacy-peer-deps` and others hadn't. It spoiled the cache pretty decently and I aimed to bypass the caching idea for the time being.

Currently, our CI tests are all finishing up within about 8 minutes max so caching isn't that big of a deal. Also I wanted to revisit caching in the GitHub Actions workflows as there were some truly silly steps going on of my own doing. I made a ticket to cover [re-evaluating / re-enabling caching](https://vajira.max.gov/browse/API-25331).

File: .github/workflows/npm-checks.yml [hunk 8]
```diff
@@ -103,20 +103,20 @@ jobs:
...
+        run: npm ci --legacy-peer-deps
```
[the remark this answers is not in the document]
Comment (gileswells) [reply .github/workflows/npm-checks.yml:119]: 🤦‍♂️

**Review by jw81 — DISMISSED** [review]
Looks good to me!  👍 
I pulled it down locally, built it, ran it, clicked all over, and it all seems to work great.

My only suggestion is in regards to [this comment](https://github.com/department-of-veterans-affairs/developer-portal/pull/1199#issuecomment-1480119640).  If it were me, I'd probably come back through and add a simple comment to those comment blocks mentioned and link to the tech debt ticket you created.  Something as simple as `# TODO - See https://vajira.max.gov/browse/API-25331`.  

That might help future-us better remember why that code is commented and remind us to come back and address it.

Just a suggestion though.  😸

**Review by imcgonigle — APPROVED** [review]
Looks good

**Review by jw81 — APPROVED** [review]

**Review by imcgonigle — APPROVED** [review]

**Pull request merged**